### PR TITLE
Update roc bert docs

### PR DESCRIPTION
**Pull request by SohamPrabhu**
# What does this PR do?
This PR updates the model card according to the changes needed on the pull request. 

#36979 (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@stevhliu 

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by SohamPrabhu** [comment]
Btw I was not sure what to put for some of the links for for the model.

File: docs/source/en/model_doc/roc_bert.md [hunk 1]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+           <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
+    </div>
+</div>
+
 # RoCBert
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[RoCBert] is a BERT model that is specifically designed for the Chinese language. It is built to resist tricks and attacks, like misspellings or similar-looking words, that usually confuse language models.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]:
```suggestion
[RoCBert](https://aclanthology.org/2022.acl-long.65.pdf) is a pretrained Chinese [BERT](./bert) model designed against adversarial attacks like typos and synonyms. It is pretrained with a contrastive learning objective to align normal and adversarial text examples. The examples include different semantic, phonetic, and visual features of Chinese. This makes RoCBert more robust against manipulation.
```

File: docs/source/en/model_doc/roc_bert.md [hunk 2]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+
+You can find all the original [Model name] checkpoints under the [RoCBert](link) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]:
```suggestion
You can find all the original RoCBert checkpoints under the [weiweishi](https://huggingface.co/weiweishi) profile.
```

File: docs/source/en/model_doc/roc_bert.md [hunk 3]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+
+> [!TIP]
+> Click on the RoCBert models in the right sidebar for more examples of how to apply RoCBert to different chinese NLP tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]:
```suggestion
> This model was contributed by [weiweishi](https://huggingface.co/weiweishi).
>
> Click on the RoCBert models in the right sidebar for more examples of how to apply RoCBert to different Chinese language tasks.

The example below demonstrates how to predict the [MASK] token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/roc_bert.md [hunk 4]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```py
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md:41]:
```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="fill-mask",
    model="weiweishi/roc-bert-base-zh",
    torch_dtype=torch.float16,
    device=0
)
pipeline("這家餐廳的拉麵是我[MASK]過的最好的拉麵之")
```

File: docs/source/en/model_doc/roc_bert.md [hunk 5]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+    model="hfl/rocbert-base"
+)
+pipeline("称呼") #Example Chinese input
+```
+
+</hfoption>
+<hfoption id="AutoModel">
+
+```py
+# pip install datasets
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]:
```py
import torch
from transformers import AutoModelForMaskedLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(
    "weiweishi/roc-bert-base-zh",
)
model = AutoModelForMaskedLM.from_pretrained(
    "weiweishi/roc-bert-base-zh",
    torch_dtype=torch.float16,
    device_map="auto",
)
inputs = tokenizer("這家餐廳的拉麵是我[MASK]過的最好的拉麵之", return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    predictions = outputs.logits

masked_index = torch.where(inputs['input_ids'] == tokenizer.mask_token_id)[1]
predicted_token_id = predictions[0, masked_index].argmax(dim=-1)
predicted_token = tokenizer.decode(predicted_token_id)

print(f"The predicted token is: {predicted_token}")
```

File: docs/source/en/model_doc/roc_bert.md [hunk 6]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+tokenizer = AutoTokenizer.from_pretrained(model_name)
+model = AutoModel.from_pretrained(model_name)
+text = "大家好，无论谁正在阅读这篇文章"
+```
+
+</hfoption>
+<hfoption id="transformers CLI">
+
+```bash
+echo -e "水在零度时会[MASK]" | transformers-cli run --task fill-mask --model junnyu/roformer_chinese_base --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]:
```suggestion
echo -e "這家餐廳的拉麵是我[MASK]過的最好的拉麵之" | transformers-cli run --task fill-mask --model weiweishi/roc-bert-base-zh --device 0
```

File: docs/source/en/model_doc/roc_bert.md [hunk 7]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
+```
+
+</hfoption>
+</hfoptions>
 
-## Overview
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]: No need to have a quantization example as its not that large and unlikely to benefit from it

File: docs/source/en/model_doc/roc_bert.md [hunk 8]
```diff
@@ -14,30 +14,76 @@ rendered properly in your Markdown viewer.
...
-This model was contributed by [weiweishi](https://huggingface.co/weiweishi).
+model = AutoModelForSequenceClassification.from_pretrained(
+    model_id,
+    quantization_config=bnb_config,
+    device_map="auto"
+)
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+```
 
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md]: Remove the `## Resources` scetion

**Review by stevhliu — COMMENTED** [review]
Thanks! You don't need to make any changes to the `moonshine.md` file :)

**Comment by SohamPrabhu** [comment]
@stevhliu Just fixed the Documentation.

File: docs/source/en/model_doc/roc_bert.md [hunk 9]
```diff
@@ -14,39 +14,77 @@ rendered properly in your Markdown viewer.
...
-- [Masked language modeling task guide](../tasks/masked_language_modeling)
-- [Multiple choice task guide](../tasks/multiple_choice)
+[RoCBert](https://aclanthology.org/2022.acl-long.65.pdf) is a pretrained Chinese [BERT](./bert) model designed against adversarial attacks like typos and synonyms. It is pretrained with a contrastive learning objective to align normal and adversarial text examples. The examples include different semantic, phonetic, and visual features of Chinese. This makes RoCBert more robust against manipulation.
+
+You can find all the original RoCBert checkpoints under the [weiweishi](https://huggingface.co/weiweishi) profile.
+
+> [!TIP]
+> This model was contributed by [weiweishi](https://huggingface.co/weiweishi).
+>
+> Click on the RoCBert models in the right sidebar for more examples of how to apply RoCBert to different Chinese language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roc_bert.md:32]:
```suggestion
> Click on the RoCBert models in the right sidebar for more examples of how to apply RoCBert to different Chinese language tasks.

The example below demonstrates how to predict the [MASK] token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/moonshine.md [hunk 10]
```diff
@@ -14,35 +14,76 @@ rendered properly in your Markdown viewer.
 
 -->
 
-# Moonshine
```
Comment (stevhliu) [inline docs/source/en/model_doc/moonshine.md]: Revert all the changes in this doc as its not needed

**Review by stevhliu — APPROVED** [review]
Thanks, just a couple more things. Remember to revert your changes to the Moonshine card

**Comment by SohamPrabhu** [comment]
@stevhliu  I changed it according to your commits, However I don't understand what I don't have in the moonshine documentation.

**Comment by stevhliu** [comment]
I think you just need to rebase your branch to get the latest changes from the Moonshine docs

**Comment by SohamPrabhu** [comment]
@stevhliu Thank you for your help I think I finally got it.
